src/compiler/checker.ts

```diff
@@ -90,6 +90,9 @@ module ts {
         let unknownType = createIntrinsicType(TypeFlags.Any, "unknown");
 
         let emptyObjectType = createAnonymousType(undefined, emptySymbols, emptyArray, emptyArray, undefined, undefined);
+        let emptyGenericType = <GenericType><ObjectType>createAnonymousType(undefined, emptySymbols, emptyArray, emptyArray, undefined, undefined);

```

Comment:
So off the bat, I actually don't know what this is used for and what exactly it is. Can you leave a nice comment giving a "visual" analogue and some intuition behind it?